lib/ts/recipe/thirdpartypasswordless/types.ts

```diff
@@ -169,12 +156,50 @@ export type PreAPIHookContext = PasswordlessPreAPIHookContext | ThirdPartyPreAPI
 export type OnHandleEventContext = PasswordlessOnHandleEventContext | ThirdPartyOnHandleEventContext;
 
 export type ThirdPartyPasswordlessSignInAndUpThemeProps = {
+    config: NormalisedConfig;
     history?: any;
+    commonState: {
+        error: string | undefined;
+    };
+
     passwordlessRecipe?: PWlessRecipe;
+    pwlessState: PWlessSignInUpState;
+    pwlessDispatch: Dispatch<PasswordlessSignInUpAction>;
+    pwlessChildProps?: PwlessChildProps;
+
     thirdPartyRecipe?: TPRecipe;
-    config: NormalisedConfig;
+    tpState: ThirdPartySignInAndUpState;
+    tpDispatch: Dispatch<ThirdPartySignInUpActions>;
+    tpChildProps?: ThirdPartySignInUpChildProps;
 };
 
+export type ThirdPartyPasswordlessSignInAndUpThemePropsWithActiveScreen = {
+    config: NormalisedConfig;
+    history?: any;
+    commonState: {
+        error: string | undefined;
+    };
+    thirdPartyRecipe?: TPRecipe;
+    tpState: ThirdPartySignInAndUpState;
+    tpDispatch: Dispatch<ThirdPartySignInUpActions>;
+    tpChildProps?: ThirdPartySignInUpChildProps;
+} & (
+    | {
+          activeScreen: undefined;
+          passwordlessRecipe: undefined;
+          pwlessState: PWlessSignInUpState;
+          pwlessDispatch: Dispatch<PasswordlessSignInUpAction>;
+          pwlessChildProps: undefined;
+      }
+    | {
+          activeScreen: SignInUpScreens;
+          passwordlessRecipe: PWlessRecipe;
+          pwlessState: PWlessSignInUpState;
+          pwlessDispatch: Dispatch<PasswordlessSignInUpAction>;
+          pwlessChildProps: PwlessChildProps;
+      }
+);
+
 export type TPPWlessRecipeInterface = {
```

Comment:
In this, consider renaming:
- `getLoginAttemptInfo` -> `getPasswordlessLoginAttemptInfo`
- `setLoginAttemptInfo` -> `setPasswordlessLoginAttemptInfo`